tests/unit/select_test.coffee

```diff
@@ -197,3 +199,55 @@ test 'Test query matching', ->
   select.set 'query', 'bana$ n'
   equal(select.get('filteredContent').length, 1,
     'duplicated spaces in the source string should be removed before matching')
+
+test "Show empty content view if content is empty", ->
+  expect 4
+
+  EmptyContentView = Ember.View.extend
+    layout: Ember.Handlebars.compile(
+      "<div class='empty-content-view'>No Content</div>"
+    )
+
+  select = @subject
+    content: []
+    optionLabelPath: 'name'
+    optionValuePath: 'code'
+    classNames: 'select-class-name'
+
+  @append()
+
+  selectElement = select.$()
+  openDropdown(selectElement)
+  andThen ->
+    ok isPresent(emptyContentSelector, selectElement),
+      'Empty content block displayed'
+    ok isNotPresent('.empty-content-view', selectElement),
+      'Empty content view not displayed before specified'
+    ok isNotPresent(noResultSelector, selectElement),
+      '"No result" message not displayed'
+  andThen ->
+    Ember.run ->
+      select.set 'emptyContentView', EmptyContentView
+  andThen ->
+    ok isPresent('.empty-content-view', selectElement),
```

Comment:
You need to make sure that setting `emptyContentView` is going to remove that view from the DOM.